packages/oruga/src/composables/useInputHandler.ts

```diff
@@ -286,6 +329,12 @@ export function useInputHandler<T extends ValidatableFormElement>(
             },
             { immediate: true },
         );
+        // The custom validation message may depend on the element's validity
+        // state, which may depend on the element's current value.
+        watch(
```

Comment:
I' not sure if this watcher is really necessary, because each component implements it's own way to check the validation state by calling `checkHtml5Validity`. 

However, if necessary, you will need to extend the `props` prop type definition of this `useInputHandler` composable.

Reply:
Are you talking about the definition [here](https://github.com/oruga-ui/oruga/pull/1023/commits/43efa277be65cbe2d39b47f34ed69c2fa35643c0#diff-e8e2be411f6a4c43f5676dd3bffe86bda2ad8c1420749d6376f37f7b8a75566dR70)?

Reply:
Ohh yeah, i haven't seen it, sorry.

Reply:
It looks like it's _almost_ possible to make all of this work without the watcher on `modelValue`, but the following sample app has issues:

```vue
<script setup lang="ts">
import { ref } from 'vue';

const val = ref<string>('');
function validation(state: ValidityState): string {
  if (state.valueMissing) { // Branch A
    return ''; // Fall back on the browser's default message in this case.
  } else { // Branch B
    return val.value.startsWith('a') ? '' : 'Enter a value that starts with "a".';
  }
}

const onSubmit = (e: Event) => { alert('Submitted'); e.preventDefault(); };
</script>

<template>
  <form @submit="onSubmit">
    <OField label="Value">
      <OInput v-model="val" required :customValidity="validation" />
    </OField>
    <OButton native-type="submit">Submit</OButton>
  </form>
</template>
```

If you enter a value that starts with the wrong character and then immediately hit Enter, the form submission goes through. Normally, the `watchEffect` call would pick up any reactive dependencies, which would include `val`. However, because the input _starts_ in an invalid state, `watchEffect` only sees Branch A the first time it runs the watch function, so it doesn't recognize the dependency on `val`. When you enter a value in the field, the element's validity state immediately changes, but because that's not a reactive value, `watchEffect` doesn't know to run the watcher again, which would run through branch B and start tracking the reactive dependency.

Reply:
Hmm maybe we can add the value to the interface of our `customValidity` prop as second or first parameter, then when the value got changed the `watchEffect` will be triggert, right?

Reply:
If we retrieve `props.modelValue` in the `watchEffect` callback, it should have the same effect as the watcher regardless of whether we pass it to `props.customValidity` or not, so that could be a viable option. The semantics would be almost exactly the same as having the watcher, but it might slightly reduce the number of duplicate calls to `updateCustomValidationMessage`. I'm slightly leaning toward just retrieving the value for the reactive dependency subscription and throwing it away because then we don't have to directly deal with defining the argument types of `props.customValidity` based on the component's `modelValue` type. How does that sound to you?

Reply:
But maybe a user would like to include the value in a custom validation message, therefore i would say adding the modelValue as attribute to the `customValidation` function might be a good solution.

Reply:
Maybe, but in basically all cases, I expect they'd have access to the model value anyway, e.g. through the `ref` they're using to drive it. The example I posted earlier is a good example; it accesses the model value without ever needing to have it passed in.